var/spack/repos/builtin/packages/tasmanian/package.py

```diff
@@ -30,7 +30,89 @@ class Tasmanian(CMakePackage):
     ApproximatioN is a robust library for high dimensional integration and
     interpolation as well as parameter calibration."""
 
-    homepage = "http://tasmanian.ornl.gov"
-    url      = "http://tasmanian.ornl.gov/documents/Tasmanian_v5.0.zip"
+    homepage = 'http://tasmanian.ornl.gov'
+    url      = 'https://github.com/ORNL/TASMANIAN/archive/v5.1.tar.gz'
 
-    version('5.0', '4bf131841d786033863d271739be0f7a')
+    version('develop', branch='master',
+            git='https://github.com/ORNL/TASMANIAN.git')
+
+    version('5.1', '5d904029a24470a6acf4a87d3339846e',
+            url='https://github.com/ORNL/TASMANIAN/archive/v5.1.tar.gz')
+
+    version('5.0', '4bf131841d786033863d271739be0f7a',
+            url='http://tasmanian.ornl.gov/documents/Tasmanian_v5.0.zip')
+
+    depends_on('cmake@3.5.0:', type='build')
+
+    variant('xsdkflags', default=False,
+            description='enable XSDK defaults for Tasmanian')
+    depends_on('python@2.7:', when='+xsdkflags', type=('build', 'run'))
+    depends_on('py-numpy', when='+xsdkflags', type=('build', 'run'))
+    extends('python', when=' +xsdkflags')
+    depends_on('mpi', when="+xsdkflags")  # openmpi 2 and 3 tested
+
+    variant('openmp', default=True,
+            description='add OpenMP support to Tasmanian')
+    # tested with OpenMP 3.1 (clang4) through 4.0-4.5 (gcc 5 - 8)
+
+    variant('blas', default=False,
+            description='add BLAS support to Tasmanian')
+    depends_on('blas', when="+blas")  # openblas 0.2.18 or newer
+
+    variant('mpi', default=False,
+            description='add MPI support to Tasmanian')
+    depends_on('mpi', when="+mpi")  # openmpi 2 and 3 tested
+
+    variant('cuda', default=False,
+            description='add CUDA support to Tasmanian')
+    depends_on('cuda@9.1', when='+cuda +magma', type=('build', 'run'))
+
+    variant('magma', default=False,
+            description='add UTK MAGMA support to Tasmanian')
+    depends_on('magma@2.3.0', when='+magma', type=('build', 'run'))
+
+    variant('python', default=False,
+            description='add Python binding for Tasmanian')
+
+    extends('python', when='+python')
+    depends_on('python@2.7:', when='+python', type=('build', 'run'))
+    depends_on('py-numpy', when='+python', type=('build', 'run'))
```

Comment:
I prefer to list all variants and then list all dependencies. When you mix them together, it becomes harder to see the full list of dependencies.